### DISCUSSION TITLE
Interpolation on unequally spaced data

### DISCUSSION THREAD

**Discussion by LeoMarHor** [ARM-software/CMSIS_5#1392] [Q&A]
Looking at the documentation page, it seems the data for linear interpolation needs to be equally spaced (https://www.keil.com/pack/doc/CMSIS/DSP/html/arm_linear_interp_example_f32_8c-example.html#a5). 
Is there any alternative for this in the library?

My data are peaks of ppg signal that can occur at any time within a predefined length window. I would need to then to interpolate them to a 4Hz grid for frequency analysis.

**Comment by christophe0606** [comment]
@LeoMarHor We have some cubic spline interpolation in the library and from the API it looks like the interpolation points can be placed as you want:

https://www.keil.com/pack/doc/CMSIS/DSP/html/group__SplineInterpolate.html

This function has no MVE acceleration so far (only pure scalar and Neon). And it is float only so I hope you don’t need fixed point arithmetic.
